checks.d/mysql.py

```diff
@@ -191,6 +193,15 @@ def _collect_metrics(self, host, db, tags, options):
                 "SHOW SLAVE STATUS", db, tags=tags
             )
 
+        # Collect custom query metrics
+        # Max of 20 queries allowed
+        if isinstance(queries, list):
+            for index, check in enumerate(queries):
+                if index == self.MAX_CUSTOM_QUERIES:
```

Comment:
Instead of checking this condition everytime, maybe we can just trim the list to at most `self.MAX_CUSTOM_QUERIES` elems before iterating: `for index, check in enumerate(queries[:self.MAX_CUSTOM_QUERIES])` The warning can be logged just based on `len(queries)`